### PR TITLE
ATT-22 AttachmentResource to check upload file size based on global p…

### PR DESCRIPTION
**Pull request by dilekamadushan**
modified the controller to check the file size.

### CODE REVIEW

File: omod-1.10/src/main/java/org/openmrs/module/attachments/rest/AttachmentResource1_10.java [hunk 1]
```diff
@@ -89,6 +89,11 @@ public Object upload(MultipartFile file, RequestContext context) throws Response
 		String fileCaption = context.getParameter("fileCaption");
 		String instructions = context.getParameter("instructions");
 		
+		// Verify File Size
+		if (attachmentsContext.getMaxUploadFileSize()<file.getSize()) {
```
Comment (mks-d) [inline omod-1.10/src/main/java/org/openmrs/module/attachments/rest/AttachmentResource1_10.java]: Spacing out:
```java
if (attachmentsContext.getMaxUploadFileSize() < file.getSize())
```

**Review by mks-d — CHANGES_REQUESTED** [review]
Test case is missing in [`AttachmentController1_10Test`](https://github.com/openmrs/openmrs-module-attachments/blob/3a8830a9f4b905e3fa69a811b27cef24b343fd14/omod-1.10/src/test/java/org/openmrs/module/attachments/rest/AttachmentController1_10Test.java).

You will have to create a new test method such as
```java
@Test
public void postAttachment_shouldNotUploadFileAboveSizeLimit() ...
```
that verifies that the expected exception is thrown when the bytes size is above the limit.

File: omod-1.10/src/main/java/org/openmrs/module/attachments/rest/AttachmentResource1_10.java (hunk 1 above)
Comment (dilekamadushan) [reply omod-1.10/src/main/java/org/openmrs/module/attachments/rest/AttachmentResource1_10.java]: @mks-d I will come up with the necessary changes. Thanks

Comment (mks-d) [reply omod-1.10/src/main/java/org/openmrs/module/attachments/rest/AttachmentResource1_10.java]: Also as @ridmal pointed out [on JIRA](https://issues.openmrs.org/browse/ATT-22?focusedCommentId=246525&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-246525) `file.getSize()` will be given in bytes, whereas `attachmentsContext.getMaxUploadFileSize()` gets back the setting in megabytes.

So you will need to apply the necessary `1024 * 1024` factor.

**Comment by dilekamadushan** [comment]
@mks-d  I modified the files accordingly. Feel free to give feedback. Thanks

File: omod-1.10/src/test/java/org/openmrs/module/attachments/rest/AttachmentController1_10Test.java [hunk 2]
```diff
@@ -234,4 +235,28 @@ public void postAttachment_shouldUploadFile() throws Exception {
...
+		
+		MockMultipartHttpServletRequest request = newUploadRequest(getURI());
+		MockMultipartFile file = new MockMultipartFile("file", fileName, "application/octet-stream", maxData);
+		
+		request.addFile(file);
+		request.addParameter("patient", patient.getUuid());
+		request.addParameter("visit", visit.getUuid());
+		request.addParameter("fileCaption", fileCaption);
+		
+		// Reply
```
Comment (mks-d) [inline omod-1.10/src/test/java/org/openmrs/module/attachments/rest/AttachmentController1_10Test.java]: `// Replay`

By the way, if you came across this typo elsewhere, would you mind correcting it there too?

File: omod-1.10/src/test/java/org/openmrs/module/attachments/rest/AttachmentController1_10Test.java [hunk 3]
```diff
@@ -234,4 +235,28 @@ public void postAttachment_shouldUploadFile() throws Exception {
...
+	public void postAttachment_shouldNotUploadFileAboveSizeLimit() throws Exception {
+		
+		String fileCaption = "Test file caption";
+		
+		// Setup
+		String fileName = "testFile1.dat";
+		Patient patient = Context.getPatientService().getPatient(2);
+		Visit visit = Context.getVisitService().getVisit(1);
+		
+		byte[] maxData = new byte[(int) ((attachmentsContext.getMaxUploadFileSize()*1024*1024)+1)];
```
Comment (mks-d) [inline omod-1.10/src/test/java/org/openmrs/module/attachments/rest/AttachmentController1_10Test.java]: Spacing in between operations:
```java
byte[] maxData = new byte[(int) ((attachmentsContext.getMaxUploadFileSize() * 1024 * 1024) + 1)];
```

File: omod-1.10/src/main/java/org/openmrs/module/attachments/rest/AttachmentResource1_10.java [hunk 4]
```diff
@@ -89,6 +89,11 @@ public Object upload(MultipartFile file, RequestContext context) throws Response
 		String fileCaption = context.getParameter("fileCaption");
 		String instructions = context.getParameter("instructions");
 		
+		// Verify File Size
+		if (attachmentsContext.getMaxUploadFileSize()*1024*1024 < (double)file.getSize()) {
```
Comment (mks-d) [inline omod-1.10/src/main/java/org/openmrs/module/attachments/rest/AttachmentResource1_10.java]: Spacing out between operations:
```java
attachmentsContext.getMaxUploadFileSize() * 1024 * 1024 < (double) file.getSize()
```

**Review by mks-d — CHANGES_REQUESTED** [review]
@dilekamadushan , this is great!

I have made very minor change requests, mostly about syntax/typos.

**Comment by dilekamadushan** [comment]
@mks-d Thank you very much for feedback.I modified the changes accordingly

**Comment by mks-d** [comment]
@dilekamadushan , just to make sure, do you run
```
mvn clean install
```
before pushing your changes to GitHub?

I'm asking because the full build takes care of code formatting.

**Comment by dilekamadushan** [comment]
@mks-d I am sorry,  I didn't do it before but I would do it from now.

**Comment by mks-d** [comment]
Also you don't need to amend + force push, you can keep pushing further commits. I will squash them eventually anyway.

File: omod-1.10/src/test/java/org/openmrs/module/attachments/rest/AttachmentController1_10Test.java [hunk 5]
```diff
@@ -234,4 +235,28 @@ public void postAttachment_shouldUploadFile() throws Exception {
 			Assert.assertNull(obs.getEncounter());
 		}
 	}
+	
+	@Test(expected = IllegalRequestException.class)
+	public void postAttachment_shouldNotUploadFileAboveSizeLimit() throws Exception {
+		
+		String fileCaption = "Test file caption";
```
Comment (mks-d) [inline omod-1.10/src/test/java/org/openmrs/module/attachments/rest/AttachmentController1_10Test.java]: Please move this under the `// Setup` comment as it's part of the test case setup.

**Review by mks-d — CHANGES_REQUESTED** [review]

**Comment by dilekamadushan** [comment]
@mks-d Thanks, actually squashing habit has been inhabited in me as it is a must when making a PR to the openmrs core

**Comment by mks-d** [comment]
@dilekamadushan fair enough, in the meantime the squashing feature was added to GitHub ;-)
It's good to bring **one commit per round of feedback** though. So that means that there is one initial commit in the PR, but further iterations can happen through subsequent commits.

Thanks for your patience and for being so reactive.

**Comment by dilekamadushan** [comment]
@mks-d You are welcome, and I must thank you back for the wonderful support you have given me. Your support is highly appreciated.

**Comment by dilekamadushan** [comment]
@mks-d Thank you very much for all the help.

**Pull request merged**